src/hera/workflows/v5/workflow.py

```diff
@@ -123,7 +125,22 @@ def _set_workflows_service(cls, v):
             return WorkflowsService()
         return v
 
+    @validator('kind', pre=True, always=True)
+    def _set_kind(cls, v):
+        if v is None:
+            return cls.__name__  # type: ignore
+        return v
+
     def build(self) -> _ModelWorkflow:
+        templates = []
+        for template in self.templates or []:
```

Comment:
`self.templates` is already defaulted to `[]` - is the `or []` necessary?

With the `templates=templates or None` used below in the `_ModelWorkflow` constructor, I'm not sure this code is the clearest as it's hiding away the `or None` below. IMO showing that templates can be `[]` or `None` up front makes the rest a bit clearer

```py
def build(self) -> _ModelWorkflow:
     templates = [] if self.templates else None
     for template in self.templates:
     # ...
```